lib/resque/worker.rb

```diff
@@ -250,9 +266,9 @@ def work(interval = 5.0, &block)
       unregister_worker(exception)
     end
 
-    def work_one_job(job = nil, &block)
+    def work_one_job(job = nil, interval: 5.0, &block)
       return false if paused?
-      return false unless job ||= reserve
+      return false unless job ||= blocking_reserve(interval)
```

Comment:
It's definitely possible, but it would be tricky to test the version Github isn't using at scale. There's some hidden complexity around `paused?`, since it can change asynchronously via sending the worker a signal.